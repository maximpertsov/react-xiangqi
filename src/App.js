import React from 'react';
import Game from './Game/Game';
import 'babel-polyfill';

// const GAME_ID = 'ABC123';

const App = () => {
  const path = window.location.pathname.replace(/^\/|\/$/g, '');

  switch (path) {
    case '':
      return <Game />;
    default:
      return <Game gameSlug={path} />;
  }
};

export default App;
