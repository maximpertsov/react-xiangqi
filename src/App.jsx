import React from 'react';
import styled from '@emotion/styled';
import Board from './components/Board/Board';
import Player from './components/Player/Player';

const Wrapper = styled.div`
  text-align: center;
`;

const App = () => (
  <Wrapper className="App">
    <Board
      redPlayer={<Player color="red" />}
      blackPlayer={<Player color="black" />}
    />
  </Wrapper>
);

export default App;
