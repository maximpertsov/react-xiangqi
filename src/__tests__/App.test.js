import React from 'react';
import { Provider } from 'react-redux';

import App from 'App';
import { render } from 'enzyme';

test('renders without crashing', () => {
  const store = mockStore({ loginForm: {}, games: [] });

  const wrapper = render(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  expect(wrapper).toMatchSnapshot();
});
