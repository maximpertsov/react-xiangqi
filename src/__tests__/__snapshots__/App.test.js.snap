// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  class="ui center aligned container"
>
  <div
    class="ui compact segments"
  >
    <div
      class="ui segment"
    >
      <div
        class="ui large header"
      >
        Play online
      </div>
      <div>
        Welcome undefined
      </div>
      <div
        class="GameList css-8bzhik"
      >
        <div
          class="ui medium header"
        >
          Games in play
        </div>
        <div
          class="css-1vrdtb6"
        />
      </div>
    </div>
    <div
      class="ui segment"
    >
      <div
        class="ui medium header"
      >
        Create game
      </div>
      <div
        class="NewGameMenu css-1hstsbv"
      >
        <div
          class="ui fluid buttons"
        >
          <button
            class="ui icon button"
          >
            <i
              aria-hidden="true"
              class="red plus square outline fitted icon"
            />
          </button>
          <button
            class="ui icon button"
          >
            <i
              aria-hidden="true"
              class="black plus square outline fitted icon"
            />
          </button>
          <button
            class="ui icon button"
          >
            <i
              aria-hidden="true"
              class="question fitted icon"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="ui segment"
    >
      <div
        class="Lobby css-8bzhik"
      >
        <div
          class="ui medium header"
        >
          Lobby
        </div>
        <div
          class="css-1vrdtb6"
        />
      </div>
    </div>
    <div
      class="ui segment"
    >
      <div
        class="ui large header"
      >
        Other modes
      </div>
      <button
        class="ui button"
      >
        Solo play
      </button>
      <button
        class="ui button"
      >
        vs CPU
      </button>
      <button
        class="ui button"
      >
        CPU vs CPU
      </button>
    </div>
  </div>
</div>
`;
